Gems/PhysX/Code/Editor/Source/Components/Conversion/PrefabConversionUtils.cpp

```diff
@@ -0,0 +1,253 @@
+
+namespace PhysX::Utils
+{
+    static AZStd::optional<AZStd::string> GetFullSourceAssetPathById(AZ::Data::AssetId assetId)
+    {
+        AZStd::string assetPath;
+        AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetPath, &AZ::Data::AssetCatalogRequests::GetAssetPathById, assetId);
+        AZ_Assert(!assetPath.empty(), "Asset Catalog returned an invalid path from an enumerated asset.");
+        if (assetPath.empty())
+        {
+            AZ_Warning(
+                "PhysXPrefabUtils",
+                false,
+                "Not able to get asset path for asset with id %s.",
+                assetId.ToString<AZStd::string>().c_str());
+            return AZStd::nullopt;
+        }
+
+        AZStd::string assetFullPath;
+        bool assetFullPathFound = false;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            assetFullPathFound,
+            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetFullSourcePathFromRelativeProductPath,
+            assetPath,
+            assetFullPath);
+        if (!assetFullPathFound)
+        {
+            AZ_Warning("PhysXPrefabUtils", false, "Source file of asset '%s' could not be found.", assetPath.c_str());
+            return AZStd::nullopt;
+        }
+
+        return { AZStd::move(assetFullPath) };
+    }
+
+    AZStd::vector<PrefabInfo> CollectPrefabs()
+    {
+        AZStd::vector<PrefabInfo> prefabs;
+
+        auto* prefabLoader = AZ::Interface<AzToolsFramework::Prefab::PrefabLoaderInterface>::Get();
+        auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
+
+        AZ::Data::AssetCatalogRequests::AssetEnumerationCB assetEnumerationCB =
+            [&prefabs, prefabLoader, prefabSystemComponent](
+                const AZ::Data::AssetId assetId, const AZ::Data::AssetInfo& assetInfo)
+        {
+            if (assetInfo.m_assetType != AzFramework::Spawnable::RTTI_Type())
+            {
+                return;
+            }
+
+            AZStd::optional<AZStd::string> assetFullPath = GetFullSourceAssetPathById(assetId);
+            if (!assetFullPath.has_value())
+            {
+                return;
+            }
+
+            if (auto templateId = prefabLoader->LoadTemplateFromFile(assetFullPath->c_str());
+                templateId != AzToolsFramework::Prefab::InvalidTemplateId)
+            {
+                if (auto templateResult = prefabSystemComponent->FindTemplate(templateId); templateResult.has_value())
+                {
+                    AzToolsFramework::Prefab::Template& templateRef = templateResult->get();
+                    prefabs.push_back({ templateId, &templateRef, AZStd::move(*assetFullPath) });
+                }
+            }
+        };
+
+        AZ::Data::AssetCatalogRequestBus::Broadcast(
+            &AZ::Data::AssetCatalogRequestBus::Events::EnumerateAssets, nullptr, assetEnumerationCB, nullptr);
```

Comment:
nitpick - would be nice to give those nullptrs names

Reply:
I'll add a variable name commented beside them.